File: src/java/org/javaturk/dp/ch05/proxy/basbakan/basbakan1/Vatandas.java

package org.javaturk.dp.ch05.proxy.basbakan.basbakan1;

public class Vatandas {
	private Basbakan basbakan = new Basbakan();

	public String derdiniAnlat() {
		basbakan.dertDinle("Bir derdim var...");
		return "Yasasiiinnn!!!";
	}

	public String isIste() {
		basbakan.isBul("Oglum");
		return "Yasasiiinnn!!!";
	}

}

File: src/go/proxy/basbakan/Vatandas.go
package basbakan


type Vatandas struct {
	basbakan Basbakan
	basbakanlikKalemi *BasbakanlikKalemi
}

func (vatandas *Vatandas) SetKalem(kalem *BasbakanlikKalemi){
	vatandas.basbakanlikKalemi = kalem
	vatandas.basbakan = vatandas.basbakanlikKalemi.BanaBasbakaniVer()
}

func (vatandas *Vatandas) DerdiniAnlat() string {
	vatandas.basbakan.DertDinle("Köyümüze lise istiyoruz.")
	return "Yasasiiinnn!!!"
}

func (vatandas *Vatandas) IsIste() string {
	vatandas.basbakan.IsBul("Oglum")
	return "Yuppiii!!!"
}

